### DISCUSSION TITLE
Sharing MATLAB experiences

### DISCUSSION THREAD

**Discussion by GitFernandoGalaz** [sampsapursiainen/zeffiro_interface#16] [Ideas]
Hello team, 

I was wondering if this can be a good idea: to share our MATLAB-related issues and complications, and describe how we solve them. This is, to enhance MATLAB coding skills :1st_place_medal:

Fernando

**Comment by sampsapursiainen** [comment]
Thumbs up if you saw this message!

**Comment by SeSodesa** [comment]
We should definitely be using function argument validation in our code, whenever possible. This not only works as code documentation, but Matlab would warn us early, if an undesired value is accidentally passed to a function. Name–value arguments can also be implemented succinctly with a Matlab `arguments` block.

See <https://se.mathworks.com/help/matlab/matlab_prog/function-argument-validation-1.html> for documentation on the subject.

**Comment by SeSodesa** [comment]
In a longer run `zef` should probably be converted to a [handle class] with property validation, and separated completely from the GUI, such that the GUI would only function as a window into what `zef` currently contains. As a handle, `zef` could emit events to inform the GUI if it needs to update a view, without the GUI itself needing to store duplicates of `zef` fields.

The structure of the program could be something like this: <https://github.com/SeSodesa/zeffiro_interface/tree/zef_as_class/%2Bzef_as_class>.

[handle class]: https://se.mathworks.com/help/matlab/handle-classes.html
